lib/Syntax/SyntaxClassifier.cpp.gyb

```diff
@@ -0,0 +1,163 @@
+%{
+  from gyb_syntax_support import *
+  # -*- mode: C++ -*-
+  # Ignore the following admonition; it applies to the resulting .cpp file only
+}%
+//// Automatically Generated From SyntaxFactory.cpp.gyb.
+//// Do Not Edit Directly!
+//===--------- SyntaxFactory.cpp - Syntax Factory implementations ---------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the SyntaxFactory, one of the most important client-facing
+// types in lib/Syntax and likely to be very commonly used.
+//
+// Effectively a namespace, SyntaxFactory is never instantiated, but is *the*
+// one-stop shop for making new Syntax nodes. Putting all of these into a
+// collection of static methods provides a single point of API lookup for
+// clients' convenience and also allows the library to hide all of the
+// constructors for all Syntax nodes, as the SyntaxFactory is friend to all.
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/Syntax/SyntaxClassifier.h"
+
+using namespace swift;
+using namespace swift::syntax;
+
+% for node in SYNTAX_NODES:
+%   if is_visitable(node):
+void SyntaxClassifier::visit(${node.name} Node) {
+%     if node.is_unknown() or node.is_syntax_collection():
+  SyntaxVisitor::visit(Node);
+%     else:
+%       for child in node.children:
+%         if child.classification:
+  visit(Node.get${child.name}(), SyntaxClassification::${child.classification}, ${"true" if child.force_classification else "false"});
+%         else:
+  visit(Node.get${child.name}());
+%         end
+%       end
+%     end
+}
+%   end
+% end
+
+/// Returns the SyntaxClassficiation a token node should receive if it is not
+/// inside a special context. Returns \c None if the token has no nativie
+/// classification and should always inherit from the context.
+llvm::Optional<SyntaxClassification>
+getNativeClassificationForToken(TokenSyntax TokenNode) {
```

Comment:
Can we rename it to `getContextFreeClassificationForToken`?

Reply:
Yeah, that name is better. Changed it.